crypto/ec/curve25519.c

```diff
@@ -5576,6 +5576,216 @@ ossl_ed25519_verify(const uint8_t *message, size_t message_len,
     return res;
 }
 
+/*
+   JM: based on hash_init_with_dom() from eddsa.c (used with Ed448).
+   There, phflag was split into two (prehashed and for_prehash) but I
+   do not understand why that was done.
+*/
+static int hash_init_with_dom(EVP_MD_CTX *hash_ctx,
+                              EVP_MD *sha512,
+                              uint8_t phflag,
+                              const uint8_t *context,
+                              size_t context_len)
+{
+#ifdef CHARSET_EBCDIC
+    const char dom_s[] = {
+        0x53, 0x69, 0x67, 0x45, 0x64, 0x32, 0x35, 0x35, 0x31, 0x39, 0x20, 0x6e,
+        0x6f, 0x20, 0x45, 0x64, 0x32, 0x35, 0x35, 0x31, 0x39, 0x20, 0x63, 0x6f,
+        0x6c, 0x6c, 0x69, 0x73, 0x69, 0x6f, 0x6e, 0x73, 0x00
+    };
+#else
+    const char dom_s[] = "SigEd25519 no Ed25519 collisions";
+#endif
```

Comment:
I think we now tend to prefer just always specifying it in hex, e.g. see:

https://github.com/openssl/openssl/blob/3c4e250eafd789bb8ff905cd2ec9dddc7e1656d2/ssl/tls13_enc.c#L416-L431

Reply:
ack.  I can change that.  Note that I was following the example inside `crypto/ec/curve448/eddsa.c`.

Do you want me to change `crypto/ec/curve448/eddsa.c` as well?

Reply:
> Do you want me to change crypto/ec/curve448/eddsa.c as well?

That would be a nice-to-have.